### PR TITLE
CloudWatch: fix variable query migration with json template variables

### LINKED ISSUE #51094: Cloudwatch variable editor crashes when migrating variable query (v9 issue)
I tried to import a dashboard with a variable query that looked like `dimension_values($region,CWAgent,cpu_usage_active,host,{"role":${role:json},"pop":${pop:json}})`  

This caused "unable to migrate poorly formed filters" errors during dashboard load and the variable query editor crashed. 

https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/cloudwatch/migrations/variableQueryMigrations.ts#L98

### PR DESCRIPTION
**Pull request by iwysiu**
**What this PR does / why we need it**:
Fixes migration for variable queries with json variables

**Which issue(s) this PR fixes**:

Fixes #51094
Fixes #51034

**Special notes for your reviewer**:
Currently it only replaces `json` values, mostly because I'm not sure of what else I would need to migrate. Would `${name}` have to be `{"env":"${name}"}` to be valid, or would users expect `{"env":${name}}` to work?

### CODE REVIEW

File: public/app/plugins/datasource/cloudwatch/datasource.ts [hunk 1]
```diff
@@ -864,8 +864,7 @@ export class CloudWatchDatasource
     });
     if (variableName && valueVar) {
       if ((valueVar as unknown as VariableWithMultiSupport).multi) {
-        // rebuild the variable name to handle old migrated queries
-        const values = this.templateSrv.replace('$' + variableName, scopedVars, 'pipe').split('|');
+        const values = this.templateSrv.replace(value, scopedVars, 'pipe').split('|');
```
Comment (sunker) [inline public/app/plugins/datasource/cloudwatch/datasource.ts:867]: Not introduced in the PR but in another PR related to variable migrations -  I think the name of this method is a bit misleading. It seems like it doesn't _only_ return the variable value, but also puts the value(s) in an array. Would it be possible to change the name and the comment to something that is a bit more descriptive? Maybe `expandVariableToArray`?

**Review by sunker — COMMENTED** [review]
Looks good! 

>Currently it only replaces json values, mostly because I'm not sure of what else I would need to migrate. Would ${name} have to be {"env":"${name}"} to be valid, or would users expect {"env":${name}} to work? 

Since the previous docs didn't specify the need to use quotes explicitly, it's probably better to migrate `${var}` to `"$var"` too. If you could add that it would be great.

**Review by aangelisc — APPROVED** [review]
LGTM👌🏽

**Review by yaelleC — APPROVED** [review]
Great tests 🚀 !

_I added a tiny comment - but it's not even on changes you made 😉_

File: public/app/plugins/datasource/cloudwatch/datasource.ts (hunk 1 above)
Comment (iwysiu) [reply public/app/plugins/datasource/cloudwatch/datasource.ts]: done

File: public/app/plugins/datasource/cloudwatch/datasource.ts [hunk 2]
```diff
@@ -864,8 +864,7 @@ export class CloudWatchDatasource
...
         return values;
```
[the remark this answers is not in the document]
Comment (iwysiu) [reply public/app/plugins/datasource/cloudwatch/datasource.ts]: done

**Review by iwysiu — COMMENTED** [review]
> Since the previous docs didn't specify the need to use quotes explicitly, it's probably better to migrate ${var} to "$var" too. If you could add that it would be great.

So I actually tried this on v8.4.x, and `${var}` only works in quotes, so we shouldn't need to migrate it

**Pull request merged**